network/some_ip/some_ip.ksy

```diff
@@ -0,0 +1,161 @@
+meta:
+  id: some_ip
+  title: AUTOSAR SOME/IP
+  license: CC0-1.0
+  ks-version: 0.9
+  endian: be
+  imports: 
+    - /network/some_ip/some_ip_sd
+
+doc: |
+  SOME/IP (Scalable service-Oriented MiddlewarE over IP) is an automotive/embedded 
+  communication protocol which supports remoteprocedure calls, event notifications 
+  and the underlying serialization/wire format.
+
+doc-ref: https://www.autosar.org/fileadmin/user_upload/standards/foundation/19-11/AUTOSAR_PRS_SOMEIPProtocol.pdf
+
+seq:
+  - id: header
+    type: header
+  - id: payload
+    size: header.length - 8
+    type:
+      switch-on: header.message_id.value
+      cases:
+        0xffff8100: some_ip_sd
+        _: data
```

Comment:
Because the `data` contains just a raw byte array stretching to the end of the stream, it's actually pointless to define it. If you leave out this line from the `cases`, this will be the implicit behavior: when `header.message_id.value != 0xffff8100`, then the `payload` effectively won't have any value of the `type` key. But omitted `type` key means byte array in KS, and this applies here as well. This case is mentioned in the [User Guide](https://doc.kaitai.io/user_guide.html#tlv).

So you can get rid of the whole `data` type completely, and it'll work the same, except that it will be accessed directly as `payload`, not `payload.body`.

```suggestion
```